engine/src/main/java/org/pentaho/di/repository/kdr/delegates/metastore/KettleDatabaseRepositoryMetaStore.java

```diff
@@ -0,0 +1,460 @@
+    try {
+      ObjectId namespaceId = delegate.getNamespaceId( namespace );
+      if ( namespaceId != null ) {
+        throw new MetaStoreNamespaceExistsException( "Namespace with name '" + namespace + "' already exists" );
+      }
+
+      // insert namespace into R_NAMESPACE
+      //
+      delegate.insertNamespace( namespace );
+      repository.commit();
+    } catch ( Exception e ) {
+      repository.rollback();
+      throw new MetaStoreException( e );
+    }
+  }
+
+  @Override
+  public void deleteNamespace( String namespace ) throws MetaStoreException, MetaStoreDependenciesExistsException {
+    try {
+      ObjectId namespaceId = delegate.verifyNamespace( namespace );
+
+      List<IMetaStoreElementType> elementTypes = getElementTypes( namespace );
+      if ( !elementTypes.isEmpty() ) {
+
+        List<String> dependencies = new ArrayList<String>();
+        for ( IMetaStoreElementType elementType : elementTypes ) {
+          dependencies.add( elementType.getId() );
+        }
+        throw new MetaStoreDependenciesExistsException( dependencies, "The namespace to delete, '"
+          + namespace + "' is not empty" );
+      }
+
+      // Now delete the namespace
+      //
+      delegate.deleteNamespace( namespaceId );
+      repository.commit();
+    } catch ( MetaStoreDependenciesExistsException e ) {
+      throw e;
+    } catch ( MetaStoreException e ) {
+      repository.rollback();
+      throw e;
+    } catch ( Exception e ) {
+      repository.rollback();
+      throw new MetaStoreException( "Unable to delete namespace '" + namespace + "'", e );
+    }
+  }
+
+  @Override
+  public boolean namespaceExists( String namespace ) throws MetaStoreException {
+    try {
+      return delegate.getNamespaceId( namespace ) != null;
+    } catch ( Exception e ) {
+      throw new MetaStoreException( "Unable to verify if namespace '" + namespace + "' exists.", e );
+    }
+  }
+
+  // Handle the element types
+
+  public void createElementType( String namespace, IMetaStoreElementType elementType ) throws MetaStoreException,
+    MetaStoreElementTypeExistsException {
+    try {
+
+      ObjectId namespaceId = delegate.verifyNamespace( namespace );
+
+      // See if the element already exists in this namespace
+      //
+      IMetaStoreElementType existingType = getElementTypeByName( namespace, elementType.getName() );
+      if ( existingType != null ) {
+        throw new MetaStoreElementTypeExistsException(
+          Arrays.asList( existingType ), "Can not create element type with id '"
+            + elementType.getId() + "' because it already exists" );
+      }
+
+      KDBRMetaStoreElementType newElementType =
+        new KDBRMetaStoreElementType( delegate, namespace, namespaceId, elementType.getName(), elementType
+          .getDescription() );
+      newElementType.save();
+      elementType.setId( newElementType.getId() );
+      repository.commit();
+    } catch ( MetaStoreElementTypeExistsException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)